app/code/community/Payone/Core/Model/Domain/Protocol/TransactionStatus.php

```diff
@@ -331,7 +331,13 @@ public function setStatusError()
     public function getRawRequestArray()
     {
         if(!empty($this->_data['raw_request'])) {
-            $aRaw = unserialize($this->_data['raw_request']);
+            $aRaw = json_decode($this->_data['raw_request'], true);
+            if (!$aRaw) { // request used to be serialized in prior versions -> try to unserialize
+                $aRaw = unserialize($this->_data['raw_request']);
```

Comment:
to avoid a Notice, you should use `@unserialize` and `@json_decode`

Reply:
`    public function getRawRequestArray()
    {
        if(!empty($this->_data['raw_request'])) {
            $aRaw = @json_decode($this->_data['raw_request'], true);
            if (!$aRaw) { // request used to be serialized in prior versions -> try to unserialize
                $aRaw = @unserialize($this->_data['raw_request']);
                if (!$aRaw) { // if unserialize didnt work, try it again with utf8_decode
                    $aRaw = @unserialize(@utf8_decode($this->_data['raw_request']));
                }
            }
            if($aRaw) {
                return $aRaw;
            }
        }

        return false;
    }`

Reply:
[code.txt](https://github.com/PAYONE-GmbH/magento-1/files/1841756/code.txt)

Reply:
We dont want to suppress notices.
There shouldn't be any notices to begin with, but if there is a problem its important to see them.

Also json_decode doesn't write notices when the string couldnt be decoded.
But you can always make those edits to the files in your project.

Reply:
you're right - looks good for me